AWSIoT/Internal/MQTTSDK/AWSMQTTSession.m

```diff
@@ -547,11 +568,11 @@ - (void)handlePuback:(AWSMQTTMessage*)msg {
     
     [[self.timerRing objectAtIndex:([flow deadline] % 60)] removeObject:msgId];
     [txFlows removeObjectForKey:msgId];
-    AWSDDLogDebug(@"Removing msgID %@ from internal store for QOS1 gaurantee", msgId);
-    [_delegate session:self newAckForMessageId:msgId.unsignedShortValue];
+    AWSDDLogDebug(@"Removing msgID %@ from internal store for QOS1 guarantee", msgId);
+    [self.delegate session:self newAckForMessageId:msgId.unsignedShortValue];
 }
 
-#pragma mark Acknowlegement Handlers for QOS 2 - not used currently as AWSIoT doesn't support QOS 2
+#pragma mark Acknowlegement Handlers for QOS 2
 - (void)handlePubrec:(AWSMQTTMessage*)msg {
```

Comment:
As above, I can't follow the call path, but please note in a comment the conditions under which this may be called in a thread-safe way.